src/measure_theory/decomposition/radon_nikodym.lean

```diff
@@ -0,0 +1,68 @@
+/-
+Copyright (c) 2021 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying
+-/
+import measure_theory.decomposition.lebesgue
+
+/-!
+# Radon-Nikodym theorem
+
+This file proves the Radon-Nikodym theorem. The Radon-Nikodym theorem states that, given measures
+`μ, ν`, if `have_lebesgue_decomposition μ ν`, then `μ` is absolutely continuous with respect to
+`ν` if and only if there exists a measurable function `f : α → ℝ≥0∞` such that `μ = fν`.
+In particular, we have `f = radon_nikodym_deriv μ ν`.
+
+The Radon-Nikodym theorem will allow us to define many important concepts in probability theory,
+most notably conditional expectations and probability cumulative functions.
+
+## Main results
+
+* `measure_theory.measureabsolutely_continuous_iff_with_density_radon_nikodym_derive_eq` :
+  the Radon-Nikodym theorem
+
+## Tags
+
+Radon-Nikodym theorem
+-/
+
+noncomputable theory
+open_locale classical measure_theory nnreal ennreal
+
+variables {α β : Type*} [measurable_space α]
```

Comment:
```suggestion
variables {α β : Type*} {m : measurable_space α}
```
Can you replace the typeclass assumption by an implicit one, to make sure it is picked from the measure?